### PR TITLE
Move to dotnet restore

### PR DESCRIPTION
**Pull request by jaredpar**
Presently we are using a mix of msbuild.exe, dotnet.exe and nuget.exe in our build. This is a lot of tools to maintain and keep in sync. This change removes nuget.exe from our build in favor of using `dotnet restore` instead. This also means that as we move forward with the SDK updates we will get updated restore behavior / warnings as well.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure 

Removed WIP as this is ready for review now.

**Comment by jaredpar** [comment]
Ping @dotnet/roslyn-infrastructure

File: build/ToolsetPackages/RoslynToolset.csproj [hunk 1]
```diff
@@ -31,7 +31,9 @@
     <PackageReference Include="RoslynTools.Microsoft.LocateVS" Version="$(RoslynToolsMicrosoftLocateVSVersion)" />
     <PackageReference Include="RoslynTools.Microsoft.SignTool" Version="$(RoslynToolsMicrosoftSignToolVersion)" />
     <PackageReference Include="RoslynTools.Microsoft.VSIXExpInstaller" Version="$(RoslynToolsMicrosoftVSIXExpInstallerVersion)" />
+    <PackageReference Include="RoslynTools.MSBuild" Version="$(RoslynToolsMSBuildVersion)" />
     <PackageReference Include="xunit.runner.wpf" Version="$(xunitrunnerwpfVersion)" />
+    <PackageReference Include="vswhere" Version="$(vswhereVersion)" />
```
Comment (tmeschter) [inline build/ToolsetPackages/RoslynToolset.csproj]: Why do we need RoslynTool.MSBuild and vswhere?

**Comment by jaredpar** [comment]
Thanks!

File: build/ToolsetPackages/RoslynToolset.csproj (hunk 1 above)
Comment (jaredpar) [reply build/ToolsetPackages/RoslynToolset.csproj]: RoslynTool.MSBuild: this is our xcopy msbuild package 
vswhere: helps us avoid downloading the xcopy msbuild package by using msbuild installed on machine if it's of a sufficient version.

**Pull request merged**